### PR TITLE
Lab 10

### PR DESCRIPTION
**Pull request by yanbixing**
Autoencoder is a an unsupervised learning of efficient data coding. With neural network layers as encoder and decoder, the aim of an autoencoder is to restore its input data at the output. Traditionally, autoencoders are used for dimensionality reduction; while recently, this concept are more widely used for learning the generative model of data. Here, we give a short summarization and some example codes of the basic class structure for autoencoders.

-- Lab 10 theory (14 Apr 2019) by Bixing Yan, Chengze Zuo. Edited by Yavuz Sunor.
-- Lab 10 coding (11 Apr 2019) by Chengze Zuo.

### CODE REVIEW

**Comment by Atcold** [comment]
PR summary, author list, and date are missing.
Plz, edit the first message, similarly to how it's done in #1.
Also, edit your folder so that this is `labs/10`, otherwise you're overwriting the previous lab.

**Comment by yanbixing** [comment]
Autoencoder is a an unsupervised learning of efficient data coding. With neural network layers as encoder and decoder, the aim of an autoencoder is to restore its input data at the output. Traditionally, autoencoders are used for dimensionality reduction; while recently, this concept are more widely used for learning the generative model of data. Here, we give a short summarization and some example codes of the basic class structure for autoencoders.

-- Lab 10 theory (14 Apr 2019) by Bixing Yan, Chengze Zuo. Edited by Yavuz Sunor.
-- Lab 10 coding (11 Apr 2019) by Chengze Zuo.

File: main.tex [hunk 1]
```diff
@@ -51,13 +52,15 @@ \part{Practice}\label{prt:practice}
...
 \input{labs/06/practice}
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \part{Coding}\label{prt:coding}
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \input{lectures/01-b/coding}
 \input{labs/01/coding}
 \input{lectures/03-a/coding}
 \input{labs/03/coding}
+\input{labs/09/coding}
```
Comment (Atcold) [inline main.tex]: You didn't rename the folder name.

File: main.tex [hunk 2]
```diff
@@ -38,6 +38,7 @@ \part{Theory}\label{prt:theory}
 \input{labs/07/theory}
 \input{lectures/07-a/optimization}
 \input{lectures/07-b/energy_based_models}
+\input{labs/09/theory.tex}
```
Comment (Atcold) [inline main.tex]: Same here. Have you actually tried to compile the LaTeX, or you expect me to fix bugs?

Comment (yanbixing) [reply main.tex]: Oops... Fixed that. Now the overleaf can compile it correctly.

**Comment by Atcold** [comment]
It's all broken!!!
Screenshot 2019-05-08 17 30 59

**Comment by Atcold** [comment]
How is that you compiled this and was working????????

**Comment by yanbixing** [comment]
> How is that you compiled this and was working????????



Really sorry for that we were using an old version. It causes some problem during merging (Like the wrong citation of Lab 09 figures). We have update our overleaf and fixed the bugs. We have created a new pull request. There is the picture of successful compiling from overleaf.

Note: However, the lecture 11 seems incomplete when we update. After we temporarily comment it. The files compiles well. 

WX20190508-210354@2x

**Pull request merged**